spec/lib/jobs/redfish-discovery-spec.js

```diff
@@ -0,0 +1,162 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+describe('Redfish Discovery Job', function () {
+    var uuid = require('node-uuid'),
+        graphId = uuid.v4(),
+        redfishJob,
+        redfishApi,
+        listChassisData,
+        getChassisData,
+        listSystemData,
+        getSystemData,
+        waterline = {},
+        sandbox = sinon.sandbox.create();
+        
+    before(function() { 
+        helper.setupInjector([
+            helper.require('/lib/jobs/base-job.js'),
+            helper.require('/lib/jobs/redfish-discovery.js'),
+            helper.di.simpleWrapper(waterline,'Services.Waterline')
+        ]);
+        waterline.nodes = {
+            findOrCreate: sandbox.stub().resolves()
+        };
+            
+        var redfish = require('redfish-node'); 
+        redfishApi = Promise.promisifyAll(new redfish.RedfishvApi());
+        
+        var Job = helper.injector.get('Job.Redfish.Discovery');
+        redfishJob = new Job({
+            uri:'fake',
+            username:'user',
+            password:'pass'
+        }, {}, graphId);
+        
+        redfishJob.redfishApi = redfishApi;
+    });
+    
+    afterEach(function() {
+        sandbox.restore();
+    });
+    
+    beforeEach(function() {
+        sandbox.stub(redfishJob.redfishApi);
+        listChassisData = [
+            null,
+            { 
+                body: {
+                    Members: [
+                        {'@odata.id':'/redfish/v1/Chassis/abc123'}
+                    ]
+                }
+            }
+        ];
+        getChassisData = [
+            null,
+            {
+                body: {
+                    Links: {
+                        ComputerSystems: [
+                            {'@odata.id':'/redfish/v1/Systems/abc123'}
+                        ]
+                    },
+                    Name: 'Chassis'
+                }
+            } 
+        ];
+        
+        listSystemData = [
+            null,
+            { 
+                body: {
+                    Members: [
+                        {'@odata.id':'/redfish/v1/Systems/abc123'}
+                    ]
+                }
+            }
+        ];
+        getSystemData = [
+            null,
+            {
+                body: {
+                    Links: {
+                        Chassis: [
+                            {'@odata.id':'/redfish/v1/Chassis/abc123'}
+                        ]
+                    },
+                    Name: 'System'
+                }
+            } 
+        ];
+    });
+    
+    describe('redfish discovery', function() {
+        it('should successfully run job', function() { 
+            redfishApi.listChassisAsync.resolves(listChassisData);
+            redfishApi.getChassisAsync.resolves(getChassisData);
+            redfishApi.listSystemsAsync.resolves(listSystemData);
+            redfishApi.getSystemAsync.resolves(getSystemData);
+            return redfishJob._run()
```

Comment:
I know it's easier to do it this way, but just FYI the way we hook into the job in the actual code is more something like:

```
redfishJob._run();
return redfishJob._deferred
.then(<...stuff...>);
```